File: apps/frontend/src/contracts/category.ts
import { z } from 'zod';

/** Taxonomy node (category or subcategory) — mirrors the API `Category`. */
export const categorySchema = z.object({
  id: z.string(),
  slug: z.string(),
  parentId: z.string().nullable(),
  labelVi: z.string(),
  labelEn: z.string(),
  icon: z.string().nullable(),
  sortOrder: z.number().int(),
});
export type Category = z.infer<typeof categorySchema>;

File: apps/backend/src/Marketplace.Domain/Categories/Category.cs
namespace Marketplace.Domain.Categories;

/// <summary>Two-level machinery taxonomy (category → subcategory) with bilingual labels.</summary>
public class Category : Entity<Guid>, IAggregateRoot
{
    public string Slug { get; private set; }

    public Guid? ParentId { get; private set; }

    public Category Parent { get; private set; }

    public ICollection<Category> Children { get; private set; } = new List<Category>();

    public string LabelVi { get; private set; }

    public string LabelEn { get; private set; }

    public string Icon { get; private set; }

    public int SortOrder { get; private set; }
}
